test/app/e2e/suites/callMessage.test.ts

```diff
@@ -59,6 +93,135 @@ describe('call', () => {
     return wasValid;
   };
 
+  const getRegExpMatch = async (regExp: RegExp) => {
+    let log = await getLog();
+    let regExpMatchGroup;
+    for (const entry of [...log].reverse()) {
+      const m = entry.content.match(regExp);
+      if (m) {
+        regExpMatchGroup = m[1];
+        break;
+      }
+    }
+    return regExpMatchGroup;
+  };
+
+  const sendValidMessageTest = async () => {
+    const callSidRegExp = /call event (CA.+): connected/;
+    const CallSid = await getRegExpMatch(callSidRegExp);
+    await axios.post(
+      `${RELAY_SERVER_URL}/create-subscription`,
+      { CallSid },
+    );
+
+    await tapButton('SEND VALID MESSAGE');
+
+    const success = await pollValidateLog((log) => {
+      const found = log.find((item) => {
+        return item.content.includes('call message sent');
+      });
+      return Boolean(found);
+    });
+
+    if (!success) {
+      throw new Error('call message sent event not received');
+    }
+
+    const voiceEventSidRegExp = /call message sent (KX.+)/;
+    const voiceEventSid = await getRegExpMatch(voiceEventSidRegExp);
+
+    const receivedMessagesResponse = await axios.get(
+      `${RELAY_SERVER_URL}/get-received-messages/${CallSid}`,
+    );
+    const receivedMessages = receivedMessagesResponse.data;
+
+    jestExpect(Array.isArray(receivedMessages)).toBeTruthy();
+    jestExpect(receivedMessages).toHaveLength(1);
+
+    const [receivedMessage] = receivedMessages;
+    jestExpect(typeof receivedMessage).toStrictEqual('object');
+    jestExpect(receivedMessage.ContentType).toStrictEqual(
+      'application/json',
+    );
+    jestExpect(receivedMessage.Content).toStrictEqual(JSON.stringify({
+      ahoy: 'This is a message from a Call',
+    }));
+    jestExpect(receivedMessage.SequenceNumber).toStrictEqual('1');
+    jestExpect(receivedMessage.CallSid).toStrictEqual(CallSid);
+    jestExpect(receivedMessage.Sid).toStrictEqual(voiceEventSid);
+  };
+
+  const sendLargeMessageTest = async () => {
+    await tapButton('SEND LARGE MESSAGE');
+
+    const success = await pollValidateLog((log) => {
+      const found = log.find((item) => {
+        return item.content.match(new RegExp(
```

Comment:
Just curious why match regex, instead of something like `.toStrictEqual` here?

Reply:
I opted for RegEx here since we can use it to capture groups, so I can pass it something like `/(CA.+)/` to parse a CallSid out of the logs (since I can't simply pass strings to the test runner from Detox).